src/dune/dune_file.ml

```diff
@@ -1976,6 +1976,43 @@ module Coqpp = struct
   type Stanza.t += T of t
 end
 
+let coq_syntax =
+  Dune_lang.Syntax.create ~name:"coq" ~desc:"the coq extension (experimental)"
+    [ ((0, 1), `Since (1, 9)); ((0, 2), `Since (2, 5)) ]
+
+module Coq_extract = struct
+  type t =
+    { extracted_modules : Module_name.t list
+    ; prelude : Loc.t * Coq_module.Name.t
+    ; flags : Ordered_set_lang.Unexpanded.t
+    ; libraries : (Loc.t * Lib_name.t) list
+    ; theories : (Loc.t * Coq_lib_name.t) list
+    ; loc : Loc.t
+    }
```

Comment:
Are these four last field common with `coq.theory` ? As mentioned above, another possibility would be to attach the extraction fields to `coq.theory` ; I guess this way is a bit more flexible tho.